composer/callbacks/checkpoint_saver.py

```diff
@@ -301,119 +301,117 @@ def __init__(
-            self._save_checkpoint(state, logger, log_level)
+            self._save_checkpoint(
+                state,
+                logger,
+                self.get_log_level(state, default=LogLevel.BATCH),
+            )
 
     def epoch_checkpoint(self, state: State, logger: Logger):
         if self.save_interval(state, Event.EPOCH_CHECKPOINT):
-            elapsed_duration = state.get_elapsed_duration()
-            assert elapsed_duration is not None, 'elapsed_duration is set on Event.BATCH_CHECKPOINT'
-            log_level = LogLevel.EPOCH if elapsed_duration < 1.0 else LogLevel.FIT
-            self._save_checkpoint(state, logger, log_level)
+            self._save_checkpoint(
+                state,
+                logger,
+                self.get_log_level(state, default=LogLevel.EPOCH),
+            )
+
+    def get_log_level(self, state: State, default: LogLevel) -> LogLevel:
+        elapsed_duration = state.get_elapsed_duration()
+        assert elapsed_duration is not None, 'elapsed_duration is set on Event.BATCH_CHECKPOINT'
+        return default if elapsed_duration < 1.0 else LogLevel.FIT
+
+    def get_state_dict(self, state):
+        return {
+            'state': state.state_dict(),
+            'rng': reproducibility.get_rng_state(),
+        }
 
     def _save_checkpoint(self, state: State, logger: Logger, log_level: LogLevel):
-        checkpoint_filepath = os.path.join(format_name_with_dist(self.folder, state.run_name), self.filename)
-        checkpoint_filepaths = checkpoint.save_checkpoint(state, checkpoint_filepath, weights_only=self.weights_only)
-
-        if dist.get_global_rank() < len(checkpoint_filepaths):
-            # Log the checkpoint as an artifact
-            checkpoint_filepath = checkpoint_filepaths[dist.get_global_rank()]
-            if self.artifact_name is not None:
-                artifact_name = format_name_with_dist_and_time(self.artifact_name, state.run_name,
-                                                               state.timestamp).lstrip('/')
-                if is_model_deepspeed(state.model) and not is_tar(artifact_name):
-                    # Deepspeed requires tarballs; appending `.tar`
-                    artifact_name += '.tar'
-                logger.file_artifact(log_level=log_level,
-                                     artifact_name=artifact_name,
-                                     file_path=checkpoint_filepath,
-                                     overwrite=self.overwrite)
-
-            if self.latest_filename is not None:
-                formatted_folder_path = format_name_with_dist(self.folder, state.run_name)
-                symlink_name = os.path.join(
-                    formatted_folder_path,
-                    format_name_with_dist_and_time(
-                        self.latest_filename,
-                        state.run_name,
-                        state.timestamp,
-                    ).lstrip('/'),
```

Comment:
should this `.lstrip('/')` be preset in `symlink` in new version?

Reply:
yes its there:
```
symlink_name = self.latest_artifact_name.format(state, is_deepspeed).lstrip('/') + '.symlink'
```